src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -109,6 +109,13 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
 #pragma warning restore CS0618 // Type or member is obsolete
                 }
 
+                if (!resolveHosts)
+                {
+                    // TODO
+                    // TODO:2 Check other places
+                    resolveHosts = connectionString.LoadBalanced;
+                }
+
```

Comment:
Why not just change line 108 to:

    resolveHosts = connectionString.Connect == ClusterConnectionMode.Direct || connectionString.Connect == ClusterConnectionMode.Standalone || connectionString.LoadBalanced;